samples/bpf/io_testing/test_protect_gpt.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+#WARNING: may overwrite GPT, use on scratch disk only
+#Tests attempts to write to various parts of disk
+#While program is attached, writes to GPT should be blocked while all other regions allowed
+
+echo "Running tests for protect_gpt bpf program"
+
+if [[ $# -ne 1 ]]; then
+	echo "Please enter device as argument"
+	exit 1
+fi
+
+TEST_DEV=$1
+GPT_SIZE=34	#number sectors in GPT
+SECTOR_SIZE=512	#bytes per sector
+
+#change to folder containing programs, 
+#protect_gpt needs to be run in same folder as protect_gpt_kern.o
+#TODO: update based on dir structure when added to xfstests
+cd ..
+
+#Load program:
+./protect_gpt $TEST_DEV protect_gpt${TEST_DEV//\//_} --attach 1> /dev/null
+
+if [[ $? -ne 0 ]]; then
+	echo "Failed to load protect_gpt program"
+	exit 1
+fi
+
+#Test: write to first GPT_SIZE blocks
+#	should fail
+dd if=/dev/zero of="$TEST_DEV" bs=$SECTOR_SIZE count=$GPT_SIZE oflag=direct &> /dev/null 
+
+if [[ $? -eq 0 ]]; then
+	echo "Failed test: program allowed writing to GPT"
+	./protect_gpt $TEST_DEV protect_gpt${TEST_DEV//\//_} --detach 1> /dev/null
+	exit 1
+fi
+
+#Test: write to last block of GPT and first block after GPT
+#	should fail
+dd if=/dev/zero of="$TEST_DEV" bs=$SECTOR_SIZE seek=$(($GPT_SIZE-1)) count=2 oflag=direct &> /dev/null
```

Comment:
Perhaps more interesting whether it fails if you issue a single IO that tries to overwrite the last block of GPT and the first block after

(You can use oflag=seek_bytes to specify the offset correctly)

Reply:
I am not sure what you mean by this, I believe the test is trying to overwrite the last block of GPT and first block after

Reply:
It is sending two separate IO requests, I'm saying it would be better to send just one that spans both blocks, i.e., count=2 bs=$((2 * $SECTOR_SIZE))

Reply:
Ah I see, if bs = $SECTOR_SIZE * 2, doesn't count just need to equal 1? I believe count is in terms of block size

Reply:
(Or feel free to leave this as a TODO comment)